Add directions for enrolling `cisagov` repositories in Lineage updates

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This pull request updates the README to add directions to enable Lineage updates for repositories with maintainers outside of our group.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

This will help any other members of the `cisagov` org get these automatic updates in their repositories if they so desire.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

Automated tests pass.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

mcdonnnj committed:
Add cisagov repository configuration blurb to the README

Add instructions on configuring a cisagov repository to receive
automatic updates.

dav3r commented on README.md:
```diff
@@ -37,6 +37,18 @@ lineage:
+If you are a [cisagov] repository maintainer and wish for your repository to
+receive Lineage updates you must do the following:
+
+1. Ensure your project is properly based on a [Skeleton repository](https://github.com/cisagov/development-guide/blob/develop/project_setup/skeleton-list.md)
```
Wording/capitalization suggestion:
```suggestion
1. Ensure your project is properly based on a [cisagov skeleton repository](https://github.com/cisagov/development-guide/blob/develop/project_setup/skeleton-list.md)
```

mcdonnnj replied:
I think I would prefer to leave the `cisagov` out simply because the configuration could be such that any other public repository could be used as the parent repository. I only link to the `cisagov` skeletons as an example.

mcdonnnj replied:
After thinking about it I decided to update the wording a bit. Please see [45ca44b](https://github.com/cisagov/action-lineage/pull/66/commits/45ca44ba74c8dfe3d724290e7432e2fa52e4712e) and let me know your thoughts.

dav3r replied:
I like it! 👍

dav3r reviewed (approved):
LGTM with one small take-it-or-leave-it suggestion. 👍

mcdonnnj committed:
Update wording in the README

Clarify that a repository should be based on any publicly accessible
parent and offer the `cisagov` skeleton repositories as options.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k commented on README.md:
```diff
@@ -37,6 +37,19 @@ lineage:
+If you are a [cisagov] repository maintainer and wish for your repository to
+receive Lineage updates you must do the following:
+
+1. Ensure your project is based on a publicly accessible parent repository
+   (such as a [`cisagov` skeleton repository](https://github.com/cisagov/development-guide/blob/develop/project_setup/skeleton-list.md))
+1. Provide [`write`](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/repository-roles-for-an-organization#repository-roles-for-organizations)
+   access to [@cisagovbot](https://github.com/cisagovbot) or [@cisagov/team-ois](https://github.com/orgs/cisagov/teams/team-ois)
```
Personally, I'd add periods to the end of these since they are complete sentences.  Take it or leave it.
```suggestion
1. Ensure your project is based on a publicly accessible parent repository
   (such as a [`cisagov` skeleton repository](https://github.com/cisagov/development-guide/blob/develop/project_setup/skeleton-list.md)).
1. Provide [`write`](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/repository-roles-for-an-organization#repository-roles-for-organizations)
   access to [@cisagovbot](https://github.com/cisagovbot) or [@cisagov/team-ois](https://github.com/orgs/cisagov/teams/team-ois).
```

jsf9k commented on README.md:
```diff
@@ -37,6 +37,19 @@ lineage:
+   access to [@cisagovbot](https://github.com/cisagovbot) or [@cisagov/team-ois](https://github.com/orgs/cisagov/teams/team-ois)
+
+Once these changes are implemented your repository should see new pull requests
+generated as appropriate during the regular Lineage organization checks.
```
Just a suggestion.  Take it or leave it; it does not impact my approval.
```suggestion
generated when appropriate during the regular Lineage organization checks.
```

jsf9k reviewed (approved):
I left a couple of _extremely_ minor suggestions, but I still approve this PR whether you take them or leave them.
